test/XRP/issued-currency-client-integration.test.ts

```diff
@@ -339,4 +339,58 @@ describe('IssuedCurrencyClient Integration Tests', function (): void {
       AccountRootFlag.LSF_NO_FREEZE,
     )
   })
+
+  it('createTrustLine - adding a trustline with 0 value', async function (): Promise<
+    void
+  > {
+    this.timeout(timeoutMs)
+    const issuer = await XRPTestUtils.randomWalletFromFaucet()
+    // GIVEN an existing testnet account and an issuer's wallet
+    // WHEN a trust line is created with the issuer with a value of 0
+    await issuedCurrencyClient.createTrustLine(
+      issuer.getAddress(),
+      'USD',
+      '0',
+      wallet,
+    )
+
+    const trustLines = await issuedCurrencyClient.getTrustLines(
+      wallet.getAddress(),
+    )
+
+    // THEN no trustlines were created.
+    assert.isArray(trustLines)
+    assert.isEmpty(trustLines)
+  })
+
+  it('createTrustLine - adding a trustline with non-zero value', async function (): Promise<
+    void
+  > {
+    this.timeout(timeoutMs)
+    const issuer = await XRPTestUtils.randomWalletFromFaucet()
+
+    const trustLineLimit = '1'
+    const trustLineCurrency = 'USD'
+
+    // GIVEN an existing testnet account and an issuer's wallet
+    // WHEN a trustline is created with the issuer with a positive value
+    await issuedCurrencyClient.createTrustLine(
+      issuer.getAddress(),
+      trustLineCurrency,
+      trustLineLimit,
+      wallet,
+    )
+
+    const trustLines = await issuedCurrencyClient.getTrustLines(
+      wallet.getAddress(),
+    )
+
+    const [createdTrustLine] = trustLines
+    const classicAddress = XrpUtils.decodeXAddress(issuer.getAddress())!
+
+    // THEN a trust line was created with the issuing account.
+    assert.equal(createdTrustLine.account, classicAddress.address)
+    assert.equal(createdTrustLine.limit, trustLineLimit)
+    assert.equal(createdTrustLine.currency, trustLineCurrency)
+  })
```

Comment:
maybe add a test attempting to create a trustline with XRP, which shouldn't be possible

Reply:
excellent idea

Reply:
done! also decided to throw an error immediately instead of waiting for the ledger to tell us we don't know what we're doing

Reply:
nice